### PR TITLE
Add dav1d/0.9.1, modernize

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **dav1d/0.9.1**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

Note: it still uses meson 0.51.0 because the upstream bug was not fixed yet. Meson version is still stored in Conan Center.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/dav1d/all/test_package/conanfile.py [hunk 1]
```diff
@@ -4,14 +4,14 @@
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    generators = "cmake", "cmake_find_package_multi"
```
Comment (madebr) [inline recipes/dav1d/all/test_package/conanfile.py:7]: Don't use the `cmake_find_package` generator here and in the test package's CMakeLists.txt.
We only test functionality that the upstream project provides.
In this case, the upstream project only provides `.pc` files, no `.cmake` scripts.

Comment (theirix) [reply recipes/dav1d/all/test_package/conanfile.py]: Done, forgot about it

**Review by SpaceIm — DISMISSED** [review]

**Review by madebr — APPROVED** [review]

**Comment by SpaceIm** [comment]
CI is sick.

**Comment by SpaceIm** [comment]
meson helper doesn't handle cross-build.

**Comment by theirix** [comment]
> meson helper doesn't handle cross-build.

I do not see any recipes with explicit cross-build stuff and meson. Should we add explicit support behind the meson helper or just deny cross-building (invalid configuration) until support will be implemented?

**Comment by madebr** [comment]
There is `conan.tools.meson.Meson` and `conan.tools.meson.MesonToolchain`.
https://docs.conan.io/en/latest/reference/conanfile/tools/meson.html

But I'm not sure that will work because it is relatively new.

**Comment by SpaceIm** [comment]
Yep: https://github.com/conan-io/conan/pull/8111

But nobody has tried to use it in CCI recipes yet. Maybe @ericLemanissier ?

**Comment by ericLemanissier** [comment]
Not yet

**Comment by SpaceIm** [comment]
I've tried to play a little bit in a simple recipe: inih

```python
from conan.tools.meson import MesonToolchain, Meson

(...)

    def generate(self):
        tc = MesonToolchain(self)
        tc.definitions["distro_install"] = True
        tc.generate()

    def _configure_meson(self):
        if self._meson:
            return self._meson
        self._meson = Meson(self)
        self._meson.configure(source_folder=self._source_subfolder)
        return self._meson

(...)
```

- native build works
- cross-build fails

```
inih/52: Calling build()
The Meson build system
Version: 0.59.0
Source dir: /Users/spaceim/.conan/data/inih/52/_/_/build/108e68814533f28957889ee7075ad46a83968698/source_subfolder
Build dir: /Users/spaceim/.conan/data/inih/52/_/_/build/108e68814533f28957889ee7075ad46a83968698/build
Build type: cross build
WARNING: Unknown options: "preprocessor_definitions"
The value of new options can be set with:
meson setup <builddir> --reconfigure -Dnew_option=new_value ...
Project name: inih
Project version: undefined

source_subfolder/meson.build:1:0: ERROR: 'c' compiler binary not defined in cross or native file
```

**Comment by madebr** [comment]
I think it needs to have `CC` explicitly defined.

**Comment by SpaceIm** [comment]
Yes but it procudes a binary for the build architecture.

**Comment by madebr** [comment]
Oh, I thought apple-clang had `gcc`/`clang` aliases.
Never mind then.

File: recipes/dav1d/all/conanfile.py [hunk 2]
```diff
@@ -46,6 +48,10 @@ def config_options(self):
             # debug builds with assembly often causes linker hangs or LNK1000
             self.options.assembly = False
 
+    def validate(self):
+        if tools.cross_building(self):
```
Comment (SpaceIm) [inline recipes/dav1d/all/conanfile.py:52]:
```suggestion
        if hasattr(self, "settings_build") and tools.cross_building(self):
```

**Review by SpaceIm — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**